lib/mobility-core/src/Kernel/Utils/Servant/Client.hs

```diff
@@ -129,10 +130,20 @@ callApiUnwrappingApiError ::
   (err -> exc) ->
   Maybe ET.ManagerSelector ->
   Maybe Text ->
+  Maybe (HM.Map BaseUrl BaseUrl) ->
   CallAPI m api a
-callApiUnwrappingApiError toAPIException mbManagerSelector errorCodeMb baseUrl eulerClient desc api =
-  callApiExtractingApiError mbManagerSelector baseUrl eulerClient desc api
-    >>= unwrapEitherCallAPIError errorCodeMb baseUrl toAPIException
+callApiUnwrappingApiError toAPIException mbManagerSelector errorCodeMb internalEndPointHashMap baseUrl eulerClient desc api = do
+  case internalEndPointHashMap of
+    Nothing ->
+      do
+        callApiExtractingApiError mbManagerSelector baseUrl eulerClient desc api
```

Comment:
move `callApiExtractingApiError` to bottom of case statement by taking url as the output of the case block